src/subdomains/generic/user/services/webhook/webhook.service.ts

```diff
@@ -0,0 +1,190 @@
+  }
+
+  // --- HELPER METHODS --- //
+
+  private async triggerWebhook(userData: UserData, data: PaymentWebhookDto | KycWebhookDto): Promise<void> {
+    userData.users = await this.userRepo.find({
+      where: { userData: { id: userData.id } },
+      relations: ['wallet', 'userData'],
+    });
+
+    for (const user of userData.users) {
+      try {
+        const currentUrl = data instanceof KycWebhookDto ? user.wallet.kycUrl : user.wallet.paymentUrl;
+
+        if (!user.wallet.isKycClient || !currentUrl) continue;
+
+        const apiKey = this.walletService.getApiKeyInternal(user.wallet.name);
+        if (!apiKey) throw new Error(`ApiKey for wallet ${user.wallet.name} not available`);
+
+        data.id = user.address;
+
+        await this.http.post(currentUrl, data, {
+          headers: { 'x-api-key': apiKey },
+          retryDelay: 5000,
+          tryCount: 3,
+        });
+      } catch (error) {
+        const errMessage = `Exception during ${data instanceof KycWebhookDto ? 'KYC' : 'Payment'} webhook for user ${
+          user.id
+        } & userData ${userData.id}:`;
+
+        console.error(errMessage, error);
+
+        await this.notificationService.sendMail({
+          type: MailType.ERROR_MONITORING,
+          input: {
+            subject: `${data instanceof KycWebhookDto ? 'KYC' : 'Payment'} Webhook failed`,
+            errors: [errMessage, error],
+          },
+        });
+      }
+    }
+  }
+
+  private async getKycWebhookData(
+    userData: UserData,
+    result: KycWebhookResult,
+    reason?: string,
+  ): Promise<KycWebhookDto> {
+    const spiderData = await this.spiderRepo.findOne({ where: { userData: { id: userData.id } } });
+
+    return {
+      result: result,
+      data: {
+        mail: userData.mail,
+        firstName: userData.firstname,
+        lastName: userData.surname,
+        street: userData.street,
+        houseNumber: userData.houseNumber,
+        city: userData.location,
+        zip: userData.zip,
+        phone: userData.phone,
+        kycStatus: this.getKycWebhookStatus(userData.kycStatus, spiderData?.chatbotResult),
+        kycHash: userData.kycHash,
+        tradingLimit: userData.tradingLimit,
+      },
+      reason: reason,
+    };
+  }
+
+  private getPaymentWebhookData(payment: BuyFiat | BuyCrypto, state: PaymentWebhookState): PaymentWebhookDto {
+    return {
+      type: payment instanceof BuyFiat ? PaymentWebhookType.BUY_FIAT : PaymentWebhookType.BUY_CRYPTO,
+      state: state,
+      inputAmount: payment.inputAmount,
+      inputAsset: payment.inputAsset,
+      outputAmount: payment.outputAmount,
+      outputAsset:
+        payment instanceof BuyFiat
+          ? payment.outputAsset
+          : payment instanceof BuyCrypto
+          ? payment.outputAsset?.name
+          : null,
+      paymentReference:
+        payment instanceof BuyFiat
+          ? payment.sell.deposit.address
```

Comment:
For FiatFiat payments we need to get the input bank usage. Maybe create a todo here?